src/pages/index.js

```diff
@@ -3,49 +3,163 @@ import Card from "../components/Card.js";
   const formList = Array.from(document.querySelectorAll(config.formSelector));
+  console.log("Forms found:", formList.length);
   formList.forEach((formElement) => {
     const validator = new FormValidator(config, formElement);
     const formName = formElement.getAttribute("name");
+    console.log("Registering validator for form:", formName);
     formValidators[formName] = validator;
     validator.enableValidation();
   });
 };
 
 enableValidation(validationSettings);
+console.log("Form validation enabled");
 
 // Initialize UserInfo
 const userInfo = new UserInfo({
   nameSelector: selectors.profileTitle,
   descriptionSelector: selectors.profileDescription,
+  avatarSelector: ".profile__image",
 });
 
+console.log("UserInfo initialized");
+
 // Initialize Section for cards
-const cardListSection = new Section(
-  {
-    items: initialCards,
-    renderer: (item) => {
-      const card = createCard(item);
-      cardListSection.addItem(card);
+let cardSection;
+let userId;
+
+function createCard(cardData) {
+  const card = new Card({
+    data: cardData,
+    cardSelector: selectors.cardTemplate,
+    handleCardClick: (data) => {
+      imagePopup.open(data);
     },
-  },
-  selectors.cardSection
-);
+    handleLikeClick: (cardId, isLiked) => {
+      console.log(
+        `Like clicked for card ${cardId}. Currently liked: ${isLiked}`
+      );
+      const likeMethod = isLiked
+        ? api.unlikeCard.bind(api)
+        : api.likeCard.bind(api);
+
+      likeMethod(cardId)
+        .then((newCardData) => {
+          console.log("New card data after like/unlike:", newCardData);
+          card.updateLikes(newCardData);
+        })
+        .catch((err) => console.error("Error updating like:", err));
+    },
+    handleDeleteClick: (cardId) => {
+      console.log("Delete clicked", cardId);
+      deleteCardPopup.open();
+      deleteCardPopup.setAction(() => {
+        deleteCardPopup.renderLoading(true);
+        api
+          .deleteCard(cardId)
+          .then(() => {
+            card.deleteCard();
+            deleteCardPopup.close();
+          })
+          .catch((err) => console.error("Error deleting card:", err))
+          .finally(() => {
+            deleteCardPopup.resetButtonState();
+          });
+      });
+    },
+    userId: userId,
+  });
+  return card.createCard();
+}
+
+function handleProfileFormSubmit(formData) {
+  console.log("Profile form submitted", formData);
+  editProfilePopup.renderLoading(true);
+  api
+    .editProfile(formData.name, formData.description)
+    .then((updatedUser) => {
+      console.log("Profile updated", updatedUser);
+      userInfo.setUserInfo({
+        name: updatedUser.name,
+        description: updatedUser.about,
+        avatar: updatedUser.avatar,
+      });
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error("Profile update error:", err);
+    })
+    .finally(() => {
+      editProfilePopup.renderLoading(false);
```

Comment:
`.renderLoading(false);` is not needed anymore (it doesn't have any effect). You have `.resetButtonText()` for that